sapi/cli/php_cli_server.c

```diff
@@ -1925,15 +1906,20 @@ static void php_cli_server_client_dtor(php_cli_server_client *client) /* {{{ */
 		client->file_fd = -1;
 	}
 	pefree(client->addr, 1);
-	pefree(client->addr_str, 1);
+	zend_string_release_ex(client->addr_str, /* persistent */ true);
+
+	/* Headers must be sent */
+	assert(client->current_header_name == NULL);
+	assert(client->current_header_value == NULL);
```

Comment:
This could be worth a test. This produces the issue:
```
$fd = stream_socket_client("tcp://localhost:8082");
fwrite($fd, "GET /index.php HTTP/1.0\r\nHost: hello"); 
fclose($fd);
```